fix(stats-fetcher): user's overall commits

Rongronggg9 opened the pull request:
1. precise overall public commits (formerly exaggerated or decreased)
2. precise overall private commits (formerly only the recent year)
1+2. precise overall commits including private commits

|current|my fork|
|-------|-------|
|![torvalds_curr]|![torvalds_my]|
|![tiwai_curr]|![tiwai_my]|
|![jhedberg_curr]|![jhedberg_my]|
|![Icenowy_curr]|![Icenowy_my]|

[torvalds_curr]: https://github-readme-stats.vercel.app/api?username=torvalds&include_all_commits=true&count_private=true
[tiwai_curr]: https://github-readme-stats.vercel.app/api?username=tiwai&include_all_commits=true&count_private=true
[jhedberg_curr]: https://github-readme-stats.vercel.app/api?username=jhedberg&include_all_commits=true&count_private=true
[Icenowy_curr]: https://github-readme-stats.vercel.app/api?username=Icenowy&include_all_commits=true&count_private=true
[torvalds_my]: https://github-readme-stats-rongronggg9.vercel.app/api?username=torvalds&include_all_commits=true&count_private=true
[tiwai_my]: https://github-readme-stats-rongronggg9.vercel.app/api?username=tiwai&include_all_commits=true&count_private=true
[jhedberg_my]: https://github-readme-stats-rongronggg9.vercel.app/api?username=jhedberg&include_all_commits=true&count_private=true
[Icenowy_my]: https://github-readme-stats-rongronggg9.vercel.app/api?username=Icenowy&include_all_commits=true&count_private=true

context: https://github.com/anuraghazra/github-readme-stats/issues/564#issuecomment-1083765940
The search API is entirely unreliable and we should never use it. Though using such a traversal could potentially cause rate limits, we should still get rid of using the search API. Why? **Almost all Linux kernel contributors could never use this project as you have already seen above. It is absolutely unacceptable, isn't it?** If we need to avoid being rate limited, we should consider using a database to cache users' commits in the past years.

Even I and @anuraghazra are affected:
|current|my fork|
|-------|-------|
|![Rongronggg9_curr]|![Rongronggg9_my]|
|![anuraghazra_curr]|![anuraghazra_my]|

[Rongronggg9_curr]: https://github-readme-stats.vercel.app/api?username=Rongronggg9&include_all_commits=true&count_private=true
[anuraghazra_curr]: https://github-readme-stats.vercel.app/api?username=anuraghazra&include_all_commits=true&count_private=true
[Rongronggg9_my]: https://github-readme-stats-rongronggg9.vercel.app/api?username=Rongronggg9&include_all_commits=true&count_private=true
[anuraghazra_my]: https://github-readme-stats-rongronggg9.vercel.app/api?username=anuraghazra&include_all_commits=true&count_private=true

If you want to check if you are affected, my deployment is https://github-readme-stats-rongronggg9.vercel.app/api

Close #518
Close #564
Close #1061
Close #1234
Close #1515

May influence #1260, #1455

ofek commented:
@anuraghazra Does this look alright to merge now?

rickstaa commented on src/fetchers/stats-fetcher.js:
```diff
@@ -63,37 +63,57 @@ const fetcher = (variables, token) => {
-    return 0;
+    return {
+      totalPublicCommits: 0,
+      totalPrivateCommits: 0,
```
Maybe we can also return a success boolean and use it in 

https://github.com/anuraghazra/github-readme-stats/blob/c03bb2f250f6d65802ca16101aa38baef1022a5c/api/index.js#L66

To set the header?

Rongronggg9 replied:
I choose to throw an error if any error occurs, just like what `fetchStats()` and `totalStarsFetcher()` do.

rickstaa replied:
Great! For your case, that is the best option since your using the GraphQL API!

Rongronggg9 committed:
fix(stats-fetcher): user's overall commits

1. precise overall public commits (formerly exaggerated)
2. precise overall private commits (formerly only the recent year)
1+2. precise overall commits including private commits

chrisK824 commented:
Shouldn't this be marked as bug and get a higher priority for merge?

rickstaa commented:
> Shouldn't this be marked as bug and get a higher priority for merge?

Although this PR is correct and makes the stats' card more correct, it will break the Public Vercel instance since it costs a lot more GraphQL points (see https://docs.github.com/en/graphql/overview/resource-limitations and https://github.com/anuraghazra/github-readme-stats/issues/1471). We therefore cannot merge this PR into the main branch.  This feature will likely be included in the new GitHub action when it is released (see https://github.com/anuraghazra/github-readme-stats/issues/2179).

mathbunnyru commented:
@Rongronggg9 @rickstaa may I ask you to fix conflicts here? I would really like to merge this PR to my fork `master` branch.

Rongronggg9 commented:
I've rebased the PR. However, due to #2100 which introduced mistaken GraphQL objects in the test suite, it cannot pass the tests. I will open another PR to fix this before I can force-push the rebased commits.

rickstaa commented:
> I've rebased the PR. However, due to #2100 which introduced mistaken GraphQL objects in the test suite, it cannot pass the tests. I will open another PR to fix this before I can force-push the rebased commits.

Weel spotted. I mreged your PR 👍🏻 !

Rongronggg9 commented:
I've rebased the PR @mathbunnyru

mathbunnyru commented:
Thanks @Rongronggg9 ❤️
